tests/std/tests/P0896R4_views_join/test.cpp

```diff
@@ -471,6 +471,38 @@ struct Immovable {
     Immovable& operator=(Immovable&&) = delete;
 };
 
+// Validate that the _Defaultabox primary template works when fed with a non-trivially-destructible type
+void test_non_trivially_destructible_type() { // COMPILE-ONLY
+    struct non_trivially_destructible_input_iterator {
+        using difference_type = int;
+        using value_type      = int;
+
+        ~non_trivially_destructible_input_iterator() {}
+
+        // To test the correct specialization of _Defaultabox, this type must not be default constructible.
+        non_trivially_destructible_input_iterator() = delete;

```

Comment:
I'm surprised that this doesn't trigger compiler warnings about "this class has no constructors" (usually I'd expect to see an `(int, int)` constructor or something similar to avoid this), but no change requested as there are apparently no test-breaking warnings.